### DISCUSSION TITLE
Translating front-end

### DISCUSSION THREAD

**Discussion by XeR** [TFNS/CTFNote#98] [Ideas]
[CTFtime stats](https://ctftime.org/stats/), as of 2021-10-25 shows :
```
US — 3882
IN — 3198
RU — 1336 (so close!)
CN — 1225
ID — 982
FR — 949
VN — 845
KR — 729
JP — 726
DE — 694
```

While US and IN speak English, this is not the case for the rest of this table.
People in my bubble speak relatively good English. I don't know if this is true for Asian teams.
I know there's a large amount of resources on zh/kr/ja websites

Do we want to translate CTFNote ?
Browsers send an `Accept-Language` header but this could be overridden in the users' settings.

Is there an "easy" way to translate the application ?
I'm thinking something like *GNU gettext*

Hopefully we can get the community to help us translate if we decide to implement this.

**Comment by JJ-8** [comment]
I don't think it is necessary to translate it to another language since English is pretty much the default language for any CTF. 

But if we want to, we have to use [App Internationalization (i18n) for Quasar](https://quasar.dev/options/app-internationalization). It uses [Vue I18n](https://kazupon.github.io/vue-i18n/introduction.html) under the hood. In order to use _GNU gettext_ with it, you have to [add another dependency](https://vuejsexamples.com/translate-your-vue-js-applications-with-gettext/) to read the `.po` files. Maybe some other software is easier to use, for example the [suggested software by Vue I18n](https://kazupon.github.io/vue-i18n/guide/tooling.html#official-tooling). [i18n Ally](https://marketplace.visualstudio.com/items?itemName=lokalise.i18n-ally) would really fit in my workflow at least :wink:

**Reply by XeR** [reply]
Hey, thanks for the reply !

> English is pretty much the default language for any CTF.

I don't know.
The problem is that, if there is a part of the community that does not speak English, we are unlikely to talk with them, by definition.
I'm hoping to get input from the users to know if such group exists, and if there is an actual need to translate CTFNote.


> In order to use GNU gettext with it, you have to add another dependency to read the .po files.

This is not a hard requirement. I mentioned *gettext* because that's what I am used to.
If Vue i18n works for our usecases, there's no need to add a layer of abstraction to get `.po` files.

Thanks a lot for the research ! :-)
